### ISSUE TITLE
Can this be adapted to generate mask conditioned histopathology patches?

### ISSUE THREAD

**Issue by bhosalems** [marcoaversa/diffinfinite#10]
Hi, we wonder if this method can be used as comparative work for our upcoming work. Can you suggest what parts need to be modified if we have to use it as a conditioned path generation model?

**Comment by marcoaversa** [comment]
Hi @bhosalems, as I told you on the other issue that you opened, since the paper’s submission, training technologies have evolved, and I’d say the training approach proposed in the paper is now somewhat outdated. However, the paper’s key strength, which remains relevant today, is the sampling method, as it doesn’t depend on how the model is trained.

We recently published a paper (linked below) along with open-source code that provides a sampling pipeline compatible with StableDiffusionXL. For your project, I’d suggest fine-tuning SDXL or SD with ControlNet and LoRA, using classes/segmentation masks/prompts as inputs and the corresponding images as targets for training. Once the model is trained, you can use the random patch denoising method found in both the DiffInfinite and Pixelsmith repositories (this new architecture) to generate infinite-resolution outputs.

If you have masks, you can train a ControlNet-based SD model to generate them from prompts, then train a second model that uses those masks as input. This workflow should give you a robust pipeline for your application.

Here you find the new work. In pixelsmith_pipeline.py you can find the random patch denoising steps in line 1747:

https://github.com/Thanos-DB/Pixelsmith

Hope this help with your project and thanks for your interest in our work.
